chia/wallet/dao_wallet/dao_wallet.py

```diff
@@ -0,0 +1,2240 @@
+        """
+        self.log.info(f"DAOWallet.coin_added() called with the coin: {coin.name()}:{coin}.")
+        wallet_node: Any = self.wallet_state_manager.wallet_node
+        peer = wallet_node.get_full_node_peer()
+        if peer is None:
+            raise ValueError("Could not find any peers to request puzzle and solution from")
+        try:
+            # Get the parent coin spend
+            cs = (await wallet_node.get_coin_state([coin.parent_coin_info], peer, height))[0]
+            parent_spend = await fetch_coin_spend(cs.spent_height, cs.coin, peer)
+
+            # check if it's a singleton and add to singleton_store
+            singleton_id = get_singleton_id_from_puzzle(parent_spend.puzzle_reveal)
+            if singleton_id:
+                await self.wallet_state_manager.singleton_store.add_spend(self.id(), parent_spend, height)
+            else:
+                # funding coin
+                asset_id = get_asset_id_from_puzzle(parent_spend.puzzle_reveal.to_program())
+                if asset_id not in self.dao_info.assets:
+                    new_asset_list = self.dao_info.assets.copy()
+                    new_asset_list.append(asset_id)
+                    dao_info = dataclasses.replace(self.dao_info, assets=new_asset_list)
+                    await self.save_info(dao_info)
+        except Exception as e:
+            self.log.error(f"Error occurred during dao wallet coin addition: {e}")
+        return
+
+    def get_cat_tail_hash(self) -> bytes32:
+        cat_wallet: CATWallet = self.wallet_state_manager.wallets[self.dao_info.cat_wallet_id]
+        return cat_wallet.cat_info.limitations_program_hash
+
+    async def adjust_filter_level(self, new_filter_level: uint64) -> None:
+        dao_info = dataclasses.replace(self.dao_info, filter_below_vote_amount=new_filter_level)
+        await self.save_info(dao_info)
+
+    async def clear_finished_proposals_from_memory(self) -> None:
+        dao_cat_wallet: DAOCATWallet = self.wallet_state_manager.wallets[self.dao_info.dao_cat_wallet_id]
+        new_list = [
+            prop_info for prop_info in self.dao_info.proposals_list
+            if not prop_info.closed
+            or prop_info.closed is None
+            or any(prop_info.proposal_id in lci.active_votes for lci in dao_cat_wallet.dao_cat_info.locked_coins)
+        ]
+        dao_info = dataclasses.replace(self.dao_info, proposals_list=new_list)
+        await self.save_info(dao_info)
+        return
+
+    async def resync_treasury_state(self) -> None:
+        parent_coin_id: bytes32 = self.dao_info.treasury_id
+        wallet_node: Any = self.wallet_state_manager.wallet_node
+        peer: WSChiaConnection = wallet_node.get_full_node_peer()
+        if peer is None:
+            raise ValueError("Could not find any peers to request puzzle and solution from")
+
+        children = await wallet_node.fetch_children(parent_coin_id, peer)
```

Comment:
why do you fetch children here?
you could just have `while True` loop and break as you are breaking now in loop below?